### DISCUSSION TITLE
就 Zonble 對威注音 Shift 中英文切換功能的不實發言一事的澄清聲明

### DISCUSSION THREAD

**Discussion by ShikiSuen** [vChewing/vChewing-macOS#118] [Announcements]
威注音輸入法的 Shift 中英文切換功能所依賴的 Shift 按鍵判定開關繼承自業火五筆輸入法。
該方案不需要額外開放 macOS 系統的輔助權限、不需要跳出 IMK 框架，也就不會存在 Zonble 所講的那種情況。
相關技術細節請洽該 Package：https://github.com/vChewing/vChewing-macOS/tree/main/Packages/Qwertyyb_ShiftKeyUpChecker

釐清事實再講話，也是一條該有的道德紅線。

P.S.: Zonble 狂言備份（只允許自己亂講，不允許對方就地澄清）：
<img width="932" alt="image" src="https://user-images.githubusercontent.com/3164826/187363851-9e8b90a7-33e0-4cd5-81a4-a801019f21a0.png">

案發現場是在上游的 @aiex718 的這個工單 https://github.com/openvanilla/McBopomofo/issues/337

**Comment by ShikiSuen** [comment]
@aiex718 感謝您講公道話。
威注音的 codebase 任您 trace，且並未用到全局鍵盤監聽（因為這種監聽是需要使用者親自手動開放才會有的）。
與 Shift 鍵盤監聽有關的程式部分已經在這篇討論串的頂樓給出 URL。

**Comment by aiex718** [comment]
@ShikiSuen 感謝您提供檔案連結，先Star此專案後續有時間再研究。

另外稍微瀏覽了一下討論區，此專案與小麥輸入法似乎有不少歷史，個人無意評論、挑起任何過往事件。
僅感謝各位開發者的心力，開發這些順手好用的輸入法。

**Reply by larryjclai** [reply]
目前用過這麼多第三方輸入法，威注音真的是我持續一直在用而沒有回到原生懷抱的一款輸入法
1. 介面好看（其他的第三方輸入法 UI 我自己比較無法適應）
2. 打字不會 LAG（我用原生的注音輸入法打字有時候會 LAG，可能敲完鍵盤都都要等個 1s ~ 2s 字詞才會出現，我是 Macbook Pro M1 2020 版本而且是 16 gb ram，不太懂原生輸入法怎麼可以這麼慘）
3. 可以用 Shift+方向鍵來隨時增減字庫很方便

非常感謝作者的貢獻，拯救了我的打字體驗！

**Reply by ShikiSuen** [reply]
@larryjclai 是您不嫌棄。

補記兩點：

1) 組字算法是 Lukhnos Liu 的，我對該算法不享有著作權。是這套頂點組字算法效率高，所以組字不會 lag。

2) Shift+方向鍵加詞的功能是小麥與奇摩輸入法所具有的，且是微軟新注音 2003 首創的功能。威注音只是允許在這種情況下降頻、刪詞而已，且威注音的這個功能對詞音不等長的情形可以原生處理＋擁有更好的 UTF8 處理。

**Comment by ShikiSuen** [comment]
@aiex718 目前的 Shift 事件偵測機制在威注音輸入法內的實作可能還有點問題。
有人抱怨過這些，只是我個人沒有遇到過就是了。
如果您有遇到的話，歡迎再開工單。

**Comment by ShikiSuen** [comment]
頂樓的檔案 URL 已經更新。
